First draft

bxin opened the pull request.

teweitsai commented on index.rst:
```diff
@@ -44,18 +44,589 @@
+The Optical Coordinate System
+#############################
+
+Our AOS system-wide CS is the Optical Coordinate System (OCS). The defition of the OCS is the same as defined in LTS-136 :cite:`LTS-136`, Section 2.2. In summary -
```
The "definition" of the OCS

teweitsai commented on index.rst:
```diff
@@ -44,18 +44,589 @@
+- LSSTCam (during commissioning and operations)
+
+Over the years, each component has been under development under their own coordinate system (CS).
+Sometimes the situation can be even more complicated - A single vendor may be using multiple CS for different sub-components (for example, mechanical versus optical). There are also inconsistencies between various engineering models and drawings.
```
may be using multiple "CSs" for

teweitsai commented on index.rst:
```diff
@@ -44,18 +44,589 @@
+- +z axis points along the optical axis, toward the sky
+- +x axis goes in parallel with the elevation axis. When look from the sky, +x points to the right.
+- +y axis can then be determined using the right hand rule. At horizon pointing, +y goes up.
+- The azimuth angle is zero when the telescope points toward northern horizon. Looking from the sky, it increase clockwise.
```
Looking from the sky, it increase"s" clockwise.

teweitsai commented on index.rst:
```diff
@@ -44,18 +44,589 @@
+- +x axis goes in parallel with the elevation axis. When look from the sky, +x points to the right.
+- +y axis can then be determined using the right hand rule. At horizon pointing, +y goes up.
+- The azimuth angle is zero when the telescope points toward northern horizon. Looking from the sky, it increase clockwise.
+- The zenith angle is zero when the telescope points at zenith. It increase to 90 degrees for horizon pointing.
```
It increase"s" to 90 degrees

teweitsai commented on index.rst:
```diff
@@ -44,18 +44,589 @@
+
+The idea is that the OCS is fixed to M1, which defines the reference for the optical system.
+Relative to the ground, the OCS rotates with the azimuth angle and the zenith angle.
+An easy way to identify the +x/+y/+z while looking at a drawing or the actual hardware is that, the walkway on the TMA, above M1M3, is in the +y direction. Otherwise when we go toward horizon it is going to bump into the ground.
```
I do not understand this sentence at all. Please help to explain more. Thanks!

bxin replied:
will add annotation to the figure to point out walkway.

teweitsai commented on index.rst:
```diff
@@ -44,18 +44,589 @@
+An easy way to identify the +x/+y/+z while looking at a drawing or the actual hardware is that, the walkway on the TMA, above M1M3, is in the +y direction. Otherwise when we go toward horizon it is going to bump into the ground.
+
+.. note::
+   The Telescope and Site top-level CS document is LTS-136 :cite:`LTS-136`, and the Camera top-level CS document is LCA-280 :cite:`LCA-280`. Unlike those two documents, we do not define the Observatory Mount CS or the Azimuth CS, because these CS, as defined by LTS-136 :cite:`LTS-136` and LCA-280 :cite:`LCA-280`, actually contradict each other, in terms of which axis points north, which points we ...
```
which points west"," etc.

teweitsai commented on index.rst:
```diff
@@ -44,18 +44,589 @@
+Zemax and PhoSim
+################
+
+We discuss the Zemax and PhoSim CS first, because these are relatively easier to define -
```
Zemax and PhoSim CS"s"

teweitsai commented on index.rst:
```diff
@@ -44,18 +44,589 @@
+
+We discuss the Zemax and PhoSim CS first, because these are relatively easier to define -
+they are self-consistent within their own framework.
+When we say Zemax CS, we refer to the global CS as used by
```
Which "global CS" you are talking about here:
https://confluence.lsstcorp.org/pages/viewpage.action?spaceKey=SYSENG&title=As-built+optical+model

bxin replied:
The confluence page links to zmx file. You will need to open the zmx file to see the CS it uses.

teweitsai replied:
Not everyone contains me has the ZEMAX license. Is it possible to explain it in the document directly? Thanks!

bxin replied:
The document already defines ZCS.

teweitsai commented on index.rst:
```diff
@@ -44,18 +44,589 @@
+       return -x,y,-z
+
+The optical sensitivity matrix (senM) is derived using the Zemax optical model.
+Therefore, everything about the senM follows the ZCS. We were able to close the simulation loop with PhoSim, because we made PhoSim consistent with Zemax.
```
Is ZCS = ZEMAX CS?

teweitsai commented on index.rst:
```diff
@@ -44,18 +44,589 @@
+
+.. note::
+
+    Note that we apply the decenters and tilts in Zemax via ``Coordinate Breaks``. Mathematically the order of decenters and tilts matter. In Zemax, there is a ``order flag``. When it is set to 0, Zemax does the decenters first, then x-tilt, y-tilt, z-rotation. When the ``order flag`` is set to 1, Zemax does these in exact opposite order, so that users can easily go back to the original CS :cite: ...
```
When you say the tilts are small enough, can you specify how small it should be to fulfill this?

bxin replied:
well, it depends on how accurate you want it to be, i.e., what your tolerance is on the difference due to the ordering. I guess one can do some simple numerical experiments to get an idea (I played with it years ago). Or, the fact that we take them as independent variables and still can make AOS converge is a proof that the ordering does not matter for practical purposes.

teweitsai commented on index.rst:
```diff
@@ -44,18 +44,589 @@
+####
+
+The M1M3 glass mirror was casted and polished at the University of Arizona Richard F. Caris Mirror Lab (RFCML).
+The mirror cell was made by CAID Industriess, and software is designed and written by the Rubin Obs. team.
```
Industries??? I am not sure this is typo or not.

teweitsai commented on index.rst:
```diff
@@ -44,18 +44,589 @@
+Like other components of the AOS, M1M3 operates mostely off its Look-Up Table (LUT), which contains our best knowledge of the forces as functions as gravity (or zenith angle) and temperature profiles on and around the mirror surfaces. The current M1M3 LUT can be found `here <https://github.com/lsst-sitcom/M1M3_ML/blob/master/data/FLUT.yaml>`__.
+
+- The zenith angle, as the primary input to the M1M3 LUT, is defined the same way as the OCS zenith angle as defined in Sec. :ref:`section-ocs`.
+- Unrelated to the bending modes, but relevant to the LUT, are the forces on the secondary cylinders of the lateral and bilateral actuators. When the force on the secondary cylinder of an lateral actuator is positive, it pushes M1M3 in the y-z plane, along 45 degrees between +y and +z axes. When the force on the secondar cylinder of a bilateral actuator is positive, it pushes M1M3 in the x-z plan ...
```
Can you specify which bilateral actuators do "+z and +x" or "+z and -x" in M1M3?

teweitsai commented on index.rst:
```diff
@@ -44,18 +44,589 @@
+   def ccs2dvcs(x,y,z):
+       return y,x,z
+
+The wavefront sensors are rotated on the focal plane. The wavefront sensor images we get from the DAQ will need to be rotated to be put into the CCS. See `here in the IM code <https://github.com/bxin/IM/blob/9d74b83eb15021e91d27bf96aa262ff378550818/source/aosWFS.py#L402>`__ or `here in ts_wep code <https://github.com/lsst-ts/ts_wep/blob/master/python/lsst/ts/wep/WepController.py#L408-L421>`__.
```
There was the discussion to unify the coordinate systems of corner WFS before (in the software CCS or DM side). If this was the case in the final, we would not need to do this rotation anymore. The decision and related work are still TBD at this moment.

bxin replied:
will add a note. thanks.

bxin commented:
Sandrine - 
Thanks for the comments. 
1. I think there should not be much confusion since the other OCS is in the software context. And I don’t want to create yet another new acronym. So I prefer to use OCS here, unless you have strong objections 🙂
2. Good idea. I added links to the bending mode plots. I didn’t include those plots in this note because this note is not really about bending modes, so I didn’t want those plots to take up too much space.
3. Same as 2.
4. I rewrote the ASC section a bit….

teweitsai commented on index.rst:
```diff
@@ -609,7 +612,13 @@ For example, in the Zemax model,
+The way the Alignment System Control (ASC) works is to measure the rigid body positions of M2 and the camera in the M1M3 CS or some CSs that are tied to the M1M3 CS, determine the offset relative to their reference positions, then command the hexapods to move M2 and the camera to their reference positions.
+
+To facilitate the above operations, it would be the easiest to have the ASC report target positions in the M2 CS and CCS, for M2 and the camera, respectively, so that the inverse of the offsets can be sent to the hexapods without further coordinate transformations.
+After a command is issued to measure the position of a target (M2 or Camera), a measurement plan is executed by the Spatial Analyzer (SA), which measures the x,y, and z of all the Spherically Mounted Reflectors (SMSs), and fit them to an internal model of the target. The CSs used by the SA are configurable.
```
"x,y, and z" --> "x, y, and z".
